auth/server/base/src/test/java/org/wildfly/security/authz/RegexRoleMapperTest.java

```diff
@@ -0,0 +1,253 @@
+
+        Iterator<String> iterator = mappedRoles.iterator();
+        int count = 0;
+        while (iterator.hasNext()) {
+            iterator.next();
+            count++;
+        }
+        assertEquals(count, 3);
+    }
+
+    @Test
+    public void testRegexMapperEmailDoNotKeep() {
+        Roles roles = createRoles("user@gmail.com", "joe", "user@customerApp.com");
+        RegexRoleMapper roleMapper = new RegexRoleMapper.Builder().setPattern(".*@([a-zA-Z]*)\\..*").setReplacement("$1-role").setKeepNonMapped(false).build();
+        Roles mappedRoles = roleMapper.mapRoles(roles);
+
+        assertFalse(mappedRoles.contains("user@gmail.com"));
+        assertFalse(mappedRoles.contains("user@customerApp.com"));
+        assertTrue(mappedRoles.contains("gmail-role"));
+        assertTrue(mappedRoles.contains("customerApp-role"));
+        assertFalse(mappedRoles.contains("joe"));
+
+        Iterator<String> iterator = mappedRoles.iterator();
+        int count = 0;
+        while (iterator.hasNext()) {
+            iterator.next();
+            count++;
+        }
+        assertEquals(count, 2);
+    }
+
+    @Test
+    public void testRegexMapperEmailDoNotKeepReplaceAll() {
+        Roles roles = createRoles("user@gmail.com", "joe", "user@customerApp.com");
+        RegexRoleMapper roleMapper = new RegexRoleMapper.Builder().setPattern(".*@([a-zA-Z]*)\\..*").setReplacement("$1-role").setKeepNonMapped(false).setReplaceAll(true).build();
+        Roles mappedRoles = roleMapper.mapRoles(roles);
+
+        assertFalse(mappedRoles.contains("user@gmail.com"));
+        assertFalse(mappedRoles.contains("user@customerApp.com"));
+        assertTrue(mappedRoles.contains("gmail-role"));
+        assertTrue(mappedRoles.contains("customerApp-role"));
+        assertFalse(mappedRoles.contains("joe"));
+
+        Iterator<String> iterator = mappedRoles.iterator();
+        int count = 0;
+        while (iterator.hasNext()) {
+            iterator.next();
+            count++;
+        }
+        assertEquals(count, 2);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testRegexMapperInvalidRegex() {
+        new RegexRoleMapper.Builder().setPattern("*-admin").setReplacement("app-$1").build();
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testRegexMapperInvalidReplacement() {
+        Roles roles = createRoles("123-admin");
+        RegexRoleMapper roleMapper = new RegexRoleMapper.Builder().setPattern(".*-admin").setReplacement("$2").build();
+        Roles mappedRoles = roleMapper.mapRoles(roles);
+        mappedRoles.contains("application-user");
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testRegexMapperInvalidReplacement2() {
+        new RegexRoleMapper.Builder().setPattern(".*-admin").setReplacement("").build();
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testRegexMapperInvalidPattern() {
```

Comment:
Maybe this could be renamed to something like testRegexMapperInvalidEmptyPattern for better readability, WDYT?

Reply:
fixed